app/server-utils/stampy.spec.ts

```diff
@@ -0,0 +1,36 @@
+import {CodaRow, loadQuestionDetail} from '~/server-utils/stampy'
+import question8486 from '~/mocks/coda-responses/question-8486.json'
+
+type ChangeFields<T, R> = Omit<T, keyof R> & R
+type CodaRowTrimmed = ChangeFields<
+  CodaRow,
+  {
+    values: Omit<
+      CodaRow['values'],
+      'Tag ID' | 'Internal?' | 'Questions tagged with this' | 'Main question' | 'Tags' // Tags shouldn't need to be ommited as it's in the data
+    >
+  }
+>
+export type questionData = [number, {items: CodaRowTrimmed[]}]
+export const questions: Array<questionData> = [[8486, question8486]]
+
+describe('loadQuestionDetail', () => {
+  it.each<questionData>(questions)('can load question %i', async (questionId, questionData) => {
+    const fetchMock = getMiniflareFetchMock()
+    fetchMock.disableNetConnect()
+
+    const origin = fetchMock.get('https://coda.io')
+    origin
+      .intercept({
+        method: 'GET',
+        path: `/apis/v1/docs/fau7sl2hmG/tables/grid-sync-1059-File/rows?useColumnNames=true&sortBy=natural&valueFormat=rich&query=%22UI%20ID%22:%22${questionId}%22`,
```

Comment:
The error message is pretty good, IMO.

I changed the `path` to be `/apis/v2` so that it doesn't match the request that `stampy.ts` makes.
The error first says:
```
TypeError: 
    >>> Error fetching https://coda.io/apis/v1/docs/fau7sl2hmG/tables/grid-sync-1059-File/rows?useColumnNames=true&sortBy=natural&valueFormat=rich&query=%22UI%20ID%22:%228486%22:
    undefined
    <<< fetch failed
```
Then after a stack trace, it says:
```
Cause:
    MockNotMatchedError: Mock dispatch not matched for path '/apis/v1/docs/fau7sl2hmG/tables/grid-sync-1059-File/rows?query=%22UI+ID%22%3A%228486%22&sortBy=natural&useColumnNames=true&valueFormat=rich': subsequent request to origin https://coda.io was not allowed (net.connect disabled)
```

To me it's pretty clear to deduce from the above that the mock path in the test needs to be changed so that is matches.

Reply:
thanks for checking, sounds good